src/types/PerformanceTable.cpp

```diff
@@ -220,36 +220,45 @@ void PerformanceTable::sort(std::string mode) {
 
 std::vector<Perf> PerformanceTable::getAltBetween(std::string critId, float inf,
                                                   float sup) {
-  if (mode_ != "crit") {
-    throw std::domain_error("Mode must be crit but is currently set to " +
-                            mode_ + ".");
-  }
+  // if (mode_ != "crit") {
+  //   throw std::domain_error("Mode must be crit but is currently set to " +
+  //                           mode_ + ".");
+  // }
   if (inf >= sup) {
     throw std::invalid_argument("Sup must be greater (>) than inf");
   }
-  if (!sorted_) {
-    throw std::domain_error("The performance table must be sorted.");
-  }
 
-  std::vector<Perf> pv = this->operator[](critId);
-  auto lower_b = std::lower_bound(
-      pv.begin(), pv.end(), inf,
-      [](const Perf &a, const float b) { return a.getValue() < b; });
-  // find the first index that have a value above sup
-  auto upper_b = std::upper_bound(
-      pv.begin(), pv.end(), sup,
-      [](const float a, const Perf &b) { return a < b.getValue(); });
-
-  if (lower_b != pv.end() && upper_b != pv.end()) {
-    return subVector(pv, std::distance(pv.begin(), lower_b),
-                     std::distance(pv.begin(), upper_b) - 1);
-  } else if (upper_b == pv.end()) {
-    return subVector(pv, std::distance(pv.begin(), lower_b), pv.size() - 1);
-  } else {
-    // return empty perf vect
-    std::vector<Perf> v;
-    return v;
+  std::vector<Perf> v;
```

Comment:
I agree, not sure what should be best though: 
- in ProfileUpdater the function is used in `crit` mode and probably in `alt` mode too (the last one depends on the altPerf table given by the model). So not sure if we can remove an unused mode.
- the way of performing the search might by faster on a single core with the check if the table is sorted, however if parallelized it might just be easier without the check for sorted. What do you think?

Reply:
If you think both modes are usefull, I'd rather implement two different functions than having two different behaviour given the mode: one with the assumption of a sorted pt `getAltBetweenSorted` for ex and one without `getAltBetween`. This would make things more readable !

I agree that with parallelization the sorted method might be obsolete, but for the moment I think we should focus on having satisfying performances on a single core! However the idea of sorting the pt before executing altBetween might not be really optimized (I haven't think that through), so it's up to you to do whatever seems best in this situation :)

Reply:
We do not have to implement it if we don't intend to use it, just trow an error if the table is not in the mode the function has been implemented for